### PR TITLE
Enable linux OOP unwinder on Windows for cross DAC

### PR DESCRIPTION
**Pull request by sdmaclea**
Add libunwind_xdac

Revise CMake files to enable compiling libunwind for UNW_REMOTE_ONLY

Add minimum set of posix prototypes and constants to enable compilation
Add minimum implementation of posix functions to satisfy linker.

Copy freebsd-elf headers from mono include directory
Cleanup comment

Fix GCC constructs not supported by VC++

Use TARGET variables to configure libunwind

Add PAL_VirtualUnwindOutOfProc to libunwind_xdac

/cc @hoyosjs

### CODE REVIEW

**Comment by janvorli** [comment]
@sdmaclea this is overall problematic. We don't want to make any changes to the libunwind sources in our repo since I expect that we would need to update them in the future from the upstream and making them diverse would cause pain once we need to sync. The exception is the CMakeLists.txt which is ours, so we can do whatever we want there.
The way we should go is to make the changes that need to be in the existing sources in the upstream libunwind. 
I would also move the files that you've added as brand new to a folder out of the libunwind (say libunwind_crossdac or something along the lines side by side with the libunwind) to keep the strong separation between the original libunwind and our private stuff.

File: src/coreclr/src/pal/src/libunwind/src/cross-dac/cross-dac.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+{
+    // 4096 is truth for most targets
+    // Unlikely to matter in dump debugging
+    return 4096;
+}
+
+void* mmap(void *addr, size_t length, int prot, int flags, int fd, size_t offset)
+{
+    // We shouldn't be doing anything other than anonymous mappings
+    if ((flags & MAP_ANON) == 0)
```
Comment (leculver) [inline src/coreclr/src/pal/src/libunwind/src/cross-dac/cross-dac.c:21]: nit:  This should be `MAP_ANONYMOUS` as `MAP_ANON` is deprecated.

File: src/coreclr/src/pal/src/libunwind/src/cross-dac/cross-dac.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        return MAP_FAILED;
+
+    return malloc(length);
```
Comment (leculver) [inline src/coreclr/src/pal/src/libunwind/src/cross-dac/cross-dac.c:24]: `MAP_ANONYMOUS` guarantees a zero-initialized buffer, shouldn't we call memset on the buffer that was malloc'ed?

File: src/coreclr/src/pal/src/libunwind/src/cross-dac/cross-dac.c [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    return 0;
+}
+
+int sigfillset(sigset_t *set)
+{
+    // For dump debugging we don't need locking
+    return 0;
+}
+
+ssize_t      read(int fd, void *buf, size_t count)
```
Comment (leculver) [inline src/coreclr/src/pal/src/libunwind/src/cross-dac/cross-dac.c:87]: nit:  Spacing?

**Comment by sdmaclea** [comment]
Closing in favor of #37521

**Pull request closed without merging**